homeassistant/components/denonavr/__init__.py

```diff
@@ -33,3 +35,40 @@ def service_handler(service):
         hass.services.register(DOMAIN, service, service_handler, schema=schema)
 
     return True
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up the denonavr components from a config entry."""
+    hass.data[DOMAIN] = {}
+
+    # Connect to receiver
+    connect_denonavr = ConnectDenonAVR(
+        hass,
+        entry.data[CONF_HOST],
+        entry.data[CONF_TIMEOUT],
+        entry.data[CONF_SHOW_ALL_SOURCES],
+        entry.data[CONF_ZONE2],
+        entry.data[CONF_ZONE3],
+    )
+    if not await connect_denonavr.async_connect_receiver():
+        return False
```

Comment:
I think it's better to use `raise ConfigEntryNotReady` than `return False`.

Reply:
Thanks for the tip